packages/loader/container-loader/src/serializedStateManager.ts

```diff
@@ -119,10 +140,93 @@ export class SerializedStateManager {
 		} else {
 			const { baseSnapshot, snapshotBlobs } = this.pendingLocalState;
 			this.snapshot = { baseSnapshot, snapshotBlobs };
+			this.refreshSnapshot(supportGetSnapshotApi).catch((error) => {
+				this.mc.logger.sendErrorEvent(
+					{
+						eventName: "Could_Not_Refresh_Snapshot",
+					},
+					error,
+				);
+				throw error;
+			});
 			return { baseSnapshot, version: undefined };
 		}
 	}
 
+	/**
+	 * Fetch latest snapshot available in storage and retry indefinetely
+	 * @param supportGetSnapshotApi -
+	 */
+	private async refreshSnapshot(supportGetSnapshotApi: boolean) {
+		// This need to be resilient to failure. We can use runWithRetry
```

Comment:
Again, this following fetch has to retry indefinitely given that we're only going to do the refresh once. If we refresh more often we could just ignore the error and addProcessedOp could retry again.

Reply:
don't worry about retry. its handled at a deeper layer, i don't think we need to do any here